twisted.internet.cfreactor not importable on Python 3.8

**Describe the incorrect behavior you saw**
fails with:

```
src/twisted/internet/cfreactor.py:474:24: error: X | Y syntax for unions requires Python 3.10  [syntax]
        _currentSimulator: object | None = None
```

```
>>> import twisted.internet.cfreactor
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/graingert/projects/twisted/src/twisted/internet/cfreactor.py", line 92, in <module>
    class CFReactor(PosixReactorBase):
  File "/home/graingert/projects/twisted/src/twisted/internet/cfreactor.py", line 474, in CFReactor
    _currentSimulator: object | None = None
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

**Describe how to cause this behavior**

import it on Python 3.8 

Preferable a [Short, Self Contained, Correct (Compilable), Example](http://www.sscce.org/) on a branch or on [a gist](https://gist.github.com).

Automated tests that are demonstrating the failure would be awesome.

**Describe the correct behavior you'd like to see**
no import error

**Testing environment**
N/A

--- a/src/twisted/internet/cfreactor.py
+++ b/src/twisted/internet/cfreactor.py
@@ -9,6 +9,7 @@ CoreFoundation main loop used by macOS.
 This is useful for integrating Twisted with U{PyObjC<http://pyobjc.sf.net/>}
 applications.
 """
+from __future__ import annotations
 
 __all__ = ["install", "CFReactor"]
 

--- a/src/twisted/newsfragments/11965.bugfix
+++ b/src/twisted/newsfragments/11965.bugfix
@@ -1,0 +1,1 @@
+Fix TypeError on t.i.cfreactor due to 3.10 type annotation syntax

